test/integration/apiserver/apply/apply_test.go

```diff
@@ -555,3 +556,191 @@ func TestApplyRemoveContainerPort(t *testing.T) {
+				}
+			},
+			"template": {
+				"metadata": {
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name":  "nginx",
+						"image": "nginx:latest"
+					}]
+				}
+			}
+		}
+	}`)
+	_, err = client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Body([]byte(obj)).Do().Get()
+	if err == nil {
+		t.Fatalf("Expecting to get version mismatch when applying object")
+	}
+	status, ok := err.(*errors.StatusError)
+	if !ok {
+		t.Fatalf("Expecting to get version mismatch as API error")
+	}
+	if status.Status().Code != http.StatusBadRequest {
+		t.Fatalf("expected status code to be %d but was %d", http.StatusBadRequest, status.Status().Code)
+	}
+}
+
+// TestCrossGroupVersionApplyManagedFields checks that the apply
+// converts the API group-version in the field manager
+func TestCrossGroupVersionApplyManagedFields(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	obj := []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "deployment",
+			"labels": {"app": "nginx"},
+			"managedFields": [
+				{
+					"manager": "create_test",
+					"operation": "Apply",
+					"apiVersion": "extensions/v1beta1",
+					"fields": {
+						"f:metadata": {
+							"f:labels": {
+								"f:app": {}
+							}
+						},
+						"f:spec": {
+							"f:replicas": {},
+							"f:selector": {
+								"f:matchLabels": {
+									"f:app": {}
+								}
+							}
+						}
+					}
+				}
+			]
+		},
+		"spec": {
+			"replicas": 3,
+			"selector": {
+				"matchLabels": {
+					 "app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name":  "nginx",
+						"image": "nginx:latest"
+					}]
+				}
+			}
+		}
+	}`)
+
+	_, err := client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Body(obj).Do().Get()
+	if err != nil {
+		t.Fatalf("Failed to create object using Apply patch: %v", err)
+	}
+
+	object, err := client.AppsV1().Deployments("default").Get("deployment", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to retrieve object: %v", err)
+	}
+
+	accessor, err := meta.Accessor(object)
+	if err != nil {
+		t.Fatalf("Failed to get meta accessor: %v", err)
+	}
+
+	managed := accessor.GetManagedFields()
+	if len(managed) != 1 && managed[0].Manager != "apply_test" {
+		t.Fatalf("Expected a single manager %v, but got managed fields: %v", "apply_test", managed)
+	}
+
+	if apiVersion := managed[0].APIVersion; apiVersion != "apps/v1" {
```

Comment:
I wish we could compare the entire output of the `managedFields`. Maybe you could build an expected managedField structure and compare it with the one that you receive? Also, you probably want some field to stay in the other apply.

Reply:
> I would maybe rename the test though to make it more clear?

Done

> I wish we could compare the entire output of the managedFields. Maybe you could build an expected managedField structure and compare it with the one that you receive?

Done -- I tried building an entire struct.

> Also, you probably want some field to stay in the other apply.

I'm actually not sure how to do this. I think I need to create a live object w/ a field manager w/ PUT first?

It seems we want a test to show that the API version is converted from `extensions/v1beta1` to `apps/v1` -- or do we simply want to check that the group version is in the field manager API version?